src/Core/Infrastructure/Security/Repository/DbWriteSecurityPolicyRepository.php

```diff
@@ -43,26 +43,28 @@ public function __construct(DatabaseConnection $db)
      */
     public function updateSecurityPolicy(SecurityPolicy $securityPolicy): void
     {
+        $configuration = json_encode([
+            "password_security_policy" => [
+                "password_length" => $securityPolicy->getPasswordMinimumLength(),
+                "has_uppercase_characters" => $securityPolicy->hasUppercase(),
+                "has_lowercase_characters" => $securityPolicy->hasLowercase(),
+                "has_numbers" => $securityPolicy->hasNumber(),
+                "has_special_characters" => $securityPolicy->hasSpecialCharacter(),
+                "attempts" => $securityPolicy->getAttempts(),
+                "blocking_duration" => $securityPolicy->getBlockingDuration(),
+                "password_expiration" => $securityPolicy->getPasswordExpiration(),
+                "delay_before_new_password" => $securityPolicy->getDelayBeforeNewPassword(),
+                "can_reuse_passwords" => $securityPolicy->canReusePasswords(),
+            ],
+        ]);
         $statement = $this->db->prepare(
             $this->translateDbName(
-                "UPDATE `:db`.password_security_policy SET " .
-                "password_length = :passwordLength, uppercase_characters = :uppercase, " .
-                "lowercase_characters = :lowercase, integer_characters = :integer, special_characters = :special, " .
-                "attempts = :attempts, blocking_duration = :blockingDuration, " .
-                "password_expiration = :passwordExpiration, delay_before_new_password = :delayBeforeNewPassword, " .
-                "can_reuse_password = :canReusePassword"
+                "UPDATE `provider_configuration`
+                SET `configuration` = :localProviderConfiguration
```

Comment:
try to use json functions to update only `password_security_policy` key
this will avoid to erase other configuration keys